crates/nu-engine/src/script.rs

```diff
@@ -77,32 +76,39 @@ pub fn process_script(
         // ...and it doesn't have any arguments
         // ...and we're in the CLI
         // ...then change to this directory
+        // We do this by renaming the command name to cd
```

Comment:
This is the main part where the logic changed.

On current main, we cd here by doing shell_manager.set_path() directly. However, this does not trigger cd site effects (like autoenv).

With this PR applied, we cd by modifying the parsed block.
The parsed block has in this case a simplified structure of:
```
CmdName: "run_external"
arg1: "directory"
```
Where as the typical `cd directory` has a structure of:
```
CmdName: "cd"
arg1: "directory"
```
So we can trigger cd here by simply changing the name.